src/fread.c

```diff
@@ -126,13 +126,11 @@ static void Field(FieldParseContext *ctx);
 //=================================================================================================
 
 /**
- * Drops `const` qualifier from a `const char*` variable, equivalent of
- * `const_cast<char*>` in C++.
+ * Drops `const` qualifier from any scalar object, equivalent of
+ * `const_cast` in C++.
  */
-static char* const_cast(const char *ptr) {
-  union { const char *a; char *b; } tmp = { ptr };
-  return tmp.b;
-}
+#define const_cast(x) (((union{typeof(x) a; typeof(+(*(x)))* b;}){.a=(x)}).b)
```

Comment:
In the issue titled 'awful one liners don't improve readability or performance' (https://github.com/Rdatatable/data.table/issues/6995) I see you wrote:

> Collapsing everything into one line doesn't make the code run faster, the compiler is smart enough to sort it out.

 I wonder what category this belongs to then 🤔